### ISSUE TITLE
2.3 Even default presentation doesn't work (default.pdf)

### ISSUE THREAD

**Issue by AndNovAtor** [bigbluebutton/bigbluebutton#10381]
**Describe the bug**
When I start "meeting" from Greenlight, there are no presentation, no slides from default.pdf. And there are nor errors in browser console, no errors in Network tab of browser console (so also no errors in nginx logs). I tried in Firefox and Chrome,

![image](https://user-images.githubusercontent.com/3628514/91987287-8c889680-ed36-11ea-9bc3-34c9ecd00049.png)

**Installation type**
Install / upgrade to last 2.3 version with bbb-install.sh
After that, I changes only few files to use SSL/https/wss (nginx file, meteor file, bbb files).
Settings for default.pdf is ... default. And should load `/var/www/bigbluebutton-default/default.pdf` (all default.* files are present there)

**Firewall and IP address type**
Local server with local IP without public access

**Additional context**
Maybe my issue is wrong or duplicate, maybe some issues are related for this, for example, https://github.com/bigbluebutton/bigbluebutton/issues/10372, https://github.com/bigbluebutton/bigbluebutton/issues/8932 - but it's JS errors, so, maybe is my issue with some server-side "component"?

Btw, presentation upload is broken, see https://github.com/bigbluebutton/bigbluebutton/issues/10372#issuecomment-685722601

**Comment by matiasilva** [comment]
Is the default.pdf file owned by the `bigbluebutton` user?

**Comment by AndNovAtor** [comment]
> Is the default.pdf file owned by the `bigbluebutton` user?
Well... no. But it's clean install. Yes, I installed from root, not from sudo user with sudo... But..
So, was something wrong on first creation of this dir? Does package installer set right permissions?

**Comment by AndNovAtor** [comment]
> `bigbluebutton` user?

I checked my working 2.2 installation - default.pdf and parrents dir have same permissions. So, looks like it's not file permissions problem (or it's new for 2.3-dev)

**Comment by schrd** [comment]
Can you download https://<hostname>/default.pdf

**Comment by AndNovAtor** [comment]
@schrd, Are you sure about root url?

**Comment by AndNovAtor** [comment]
> Can you download https://$hostname/default.pdf

Yes, it's oppening. Same as with 2.2

**Comment by schrd** [comment]
> @schrd, Are you sure about root url, /default.pdf?

yes. It is fetched from localhost and then converted to SVG AFAIK which is then delivered to the browser. From my logs:
```
127.0.0.1 - - [01/Sep/2020:18:17:56 +0200] "GET /default.pdf HTTP/1.1" 200 257231 "-" "Mozilla" rt="0.000" uct="-" uht="-" urt="-"
127.0.0.1 - - [01/Sep/2020:18:17:57 +0200] "GET /default.pdf HTTP/1.1" 200 257231 "-" "Apache-HttpAsyncClient/4.1.4 (Java/1.8.0_265)" rt="0.000" uct="-" uht="-" urt="-"
```

**Comment by AndNovAtor** [comment]
> yes. It is fetched from localhost and then converted to SVG 

Yes, so browser downloads SVG, not PDF.

So.... this svg have some path in  `/var/bigbluebutton/<some id1>/<some id1>/some another id2/svgs` for 2.2.
![image](https://user-images.githubusercontent.com/3628514/92125600-e0af7b80-ee07-11ea-81d0-ff6329416549.png)

Now, I opened same dir for 2.3-dev - and it's empty.
![image](https://user-images.githubusercontent.com/3628514/92125619-e4db9900-ee07-11ea-81f9-26bb18d4bf0d.png)

I compared file permissions and owners for dir for 2.2 and 2.3-dev - they are the same: `755`, `bigbluebutton:bigbluebutton`

I remember, that convert mechanism was broken for non-pdf files,.... so is it broken now also for pdf?

**Comment by AndNovAtor** [comment]
@schrd, Are you have installed greenlight with 2.3-dev (and default.pdf works)?

**Comment by schrd** [comment]
yes, Greenlight 2.7.1 with Shibboleth patches (runs on different machine). Greenlight is configured not to allow preloading of PDFs to the rooms. The BBB 2.3 machine is deployed with using the same recipes like our 2.2 machines. We deploy a customized pdf, not the one shipped with BBB.

**Comment by AndNovAtor** [comment]
Meh, I'm lucky.

Btw, I tried another pdf or same pdf with another name (and changed config for this path) - same (fail) result

**Comment by AndNovAtor** [comment]
Idk.... I upgrade clean Ubuntu 18.04 up to date. I install  AGAIN ("clean") BBB 2.3-dev with Greenlight by latest bbb-install.sh. Create admin user, login.
Start "room" - and AGAIN there no presentation...  Presentation dir `/var/bigbluebutton/<some id1>/<some id1>/some another id2` is emty again... What I doing wrong?...

**Comment by AndNovAtor** [comment]
Hm, there are NO Libreoffice O_. So bbb-conf --restart also return "dpkg-query: no package found matching libreoffice" O_

**Comment by schrd** [comment]
libreoffice runs in a docker container. You need to have docker enabled and running before bbb-install.sh is started

**Comment by AndNovAtor** [comment]
@schrd , ok, so it's will be time for separate issue - bbb-conf still runs dpkg-query, and of course it returns warning that there are no libreoffice...

About docker. bbb-install.sh installs docker BEFORE all bbb installs, so there are no need install docker BEFORE. Otherwice I even can make docker installations conflict 

```
...
+ add-apt-repository 'deb [arch=amd64] https://download.docker.com/linux/ubuntu      bionic      stable'
...
+ need_pkg docker-ce
....
+ apt-get install -yq docker-ce
...
The following NEW packages will be installed:
  aufs-tools cgroupfs-mount containerd.io docker-ce docker-ce-cli libltdl7
  pigz

```

**Comment by AndNovAtor** [comment]
OK, I found the reason. Yes, it's partly only my problem, but there are some interesting things.

Ok, presentation SVG/i.e. "generation" not work, if I set up ssl certs and https, but Greenlight endpoint is STILL http. I did it, because I use custom CA and SSL cert signed with it (CA cert added to system, but not Greenlight docker), and Chrome works only with wss (and only https needed for microphone input).

Interesting, that all other things works with this "setup". Nginx listens 80 and 443 for BBB.
(So, I chaged only ws to wss in meteor and set up ssl in Nginx - then presentation works =) )

**Comment by AndNovAtor** [comment]
Btw, @schrd , do you know a way to add files to currently exist Greenlight docker container? I need add my CA files to Greenlight container and run ```update-ca-certificates```. I know that I can build Greenlight docker container from source, and add files on building, but... Docker use "volumes", also support ```docker exec```.. I can add files and run commands withiout building Greenlight from source, can't I?

(So I strongly upvotes for some feauture in Greenlight to add custom certs... but it for Greenlight issues, for example, https://github.com/bigbluebutton/greenlight/issues/1642 and https://github.com/bigbluebutton/greenlight/issues/1455)

**Comment by schrd** [comment]
you can build your own container by inheriting from bigbluebutton/greenlight:v2 and then adding the commands and files to import certificates. I think github issues is not the right place for discussing this. It seems to me that the original issue does not exist in BBB, so you might close this issue to prevent open issues in BBB piling up.

**Comment by AndNovAtor** [comment]
@schrd , I already wrote that I know about inheriting from bigbluebutton/greenlight:v2...
Anyway, I close this issue HERE (maybe create some issue in greenlight), because it seems like Greenlight specific, so of cource it's wrong place.

**Comment by lonesomewalker** [comment]
Sorry to dig this up, but we also have this now on 2 servers.
Setup as i have seen so far with bbb-install.sh, no self-signed certificates. LE default settings.'
default.pdf is available for the hostname from outside.

Seems there is something broken inside :-/

**Comment by phant** [comment]
Today i have same issue with presentation.

Maybe problem issued by Let's Encrypt (DST Root CA X3 Expiration).

BigBlueButton Server 2.2.36 (2416)
Kernel version: 4.15.0-142-generic
Distribution: Ubuntu 16.04.7 LTS (64-bit)


![2021-10-01_13-29-18](https://user-images.githubusercontent.com/805222/135596271-87e21378-a469-4b1a-8380-5a0c0020bb5c.png)

**Comment by GhaziTriki** [comment]
For 2.2 https://groups.google.com/g/bigbluebutton-setup/c/U3Xj_hQSe5s/

**Comment by AndNovAtor** [comment]
> Today i have same issue with presentation.
> 
> Maybe problem issued by Let's Encrypt (DST Root CA X3 Expiration).

As all sayed, it's affected ubuntu 16.04. 

Tldr:
https://letsencrypt.org/certificates/
![image](https://user-images.githubusercontent.com/3628514/135608217-861ec1cb-6dcc-42d2-be46-a08ff14f26eb.png)

So 2 ways:
1) as wroted on groups - force renew cert to use withous crooss-chain to use only valid ISRG Root X1
2) as wroted before on some pages ("regardless" of bug in openssl prior 1.0.2) - manually remove DST Root CA X3 from trusted CA "storage" and import official "self-signed" ISRG Root X1

**Comment by enyor** [comment]
> Today i have same issue with presentation.
> 
> Maybe problem issued by Let's Encrypt (DST Root CA X3 Expiration).
> 
> BigBlueButton Server 2.2.36 (2416) Kernel version: 4.15.0-142-generic Distribution: Ubuntu 16.04.7 LTS (64-bit)
> 
> ![2021-10-01_13-29-18](https://user-images.githubusercontent.com/805222/135596271-87e21378-a469-4b1a-8380-5a0c0020bb5c.png)

If you installed greenligth and redirect the route https: //youserver/default.pdf stops working so that redirection is the problem, you must indicate in the configuration a route that can be accessed
